custom_components/modbus_local_gateway/tcp_client.py

```
@@ -18,6 +19,17 @@
 _LOGGER: logging.Logger = logging.getLogger(__name__)
 
 
+class MyTransactionManager(TransactionManager):
+    """Custom Transaction Manager to supress exception logging"""
+
+    def data_received(self, data: bytes) -> None:
+        """Catch any protocol exceptions so they don't pollute the HA logs"""
+        try:
+            super().data_received(data)
+        except ModbusIOException:
+            pass
```

Ruff (0.8.2)

Use `contextlib.suppress(ModbusIOException)` instead of `try`-`except`-`pass`

Replace with `contextlib.suppress(ModbusIOException)`

(SIM105)